aiohttp/web_protocol.py

```diff
@@ -164,8 +189,11 @@ def __init__(self, manager: 'Server', *,
         self.debug = debug
         self.access_log = access_log
         if access_log:
-            self.access_logger = access_log_class(
-                access_log, access_log_format)  # type: Optional[AbstractAccessLogger]  # noqa
+            if issubclass(access_log_class, AbstractAsyncAccessLogger):
```

Comment:
`web_protocol.RequestHandler` is a private class, a part of implementation details.
Would you consider moving access log initialization into `web.Server`/`web.Application`.
`RequestHandler.__init__` signature can be changed in a wild, e.g. replacement all `access_log_class`, `access_log`, `access_log_format` with access logger instance is totally accepted.

I still have no clear vision of how exactly it should look like, and probably cannot have without prototyping.
If you want to play with the idea -- let me know. 

Othervise the PR looks good to me.